Detectors/Upgrades/ALICE3/FCT/simulation/src/FCTLayer.cxx

```diff
@@ -0,0 +1,202 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file FCTLayer.cxx
+/// \brief Implementation of the FCTLayer class
+/// \author Mario Sitta <sitta@to.infn.it>
+/// \author Chinorat Kobdaj (kobdaj@g.sut.ac.th)
+
+#include "FCTSimulation/FCTLayer.h"
+#include "FCTBase/GeometryTGeo.h"
+#include "FCTSimulation/Detector.h"
+
+#include "FairLogger.h" // for LOG
+
+#include <TGeoManager.h>        // for TGeoManager, gGeoManager
+#include <TGeoMatrix.h>         // for TGeoCombiTrans, TGeoRotation, etc
+#include <TGeoTube.h>           // for TGeoTube, TGeoTubeSeg
+#include <TGeoVolume.h>         // for TGeoVolume, TGeoVolumeAssembly
+#include <TGeoCompositeShape.h> // for TGeoCompositeShape
+#include "TMathBase.h"          // for Abs
+#include <TMath.h>              // for Sin, RadToDeg, DegToRad, Cos, Tan, etc
+
+#include <cstdio> // for snprintf
+
+class TGeoMedium;
+
+using namespace TMath;
+using namespace o2::fct;
+using namespace o2::itsmft;
+
+ClassImp(FCTLayer);
+
+FCTLayer::~FCTLayer() = default;
```

Comment:
I reckon that should go to the header file

Reply:
I tried, but the building failed afterwards.